DBZ-2049 Add deployment steps to the doc for each connector

TovaCohen opened the pull request:
These updates are for [DBZ-2049](https://issues.redhat.com/browse/DBZ-2049).
I used the deployment instructions that we updated in the PostgreSQL connector doc as the model to update and add deployment instructions for each connector. 
There is a separate commit for the updates in each connector. 
I manually copied the updated files downstream to make sure that the downstream book builds. You might notice a few updates that seem unrelated to this update, but I had to make them to get the book to build. 
In the PostgreSQL connector doc, I updated a link target URL. 
I confirmed that Antora builds the doc. 
These updates are not required for 1.2, but it would be nice to be able to provide them in the downstream 1.2 doc. 
Let me know what needs to change.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mongodb.adoc:
```diff
@@ -594,60 +594,58 @@ Following is an example of what a message looks like:
+extract the JAR files into your Kafka Connect environment, and add the directory with the JAR files to  {link-kafka-docs}/#connectconfigs[Kafka Connect's `plugin.path`]. You then need to 
+restart your Kafka Connect process to pick up the new JAR files.
+
+If you need immutable containers, see link:https://hub.docker.com/r/debezium/[{prodname}'s Container images] for Zookeeper, Kafka, and Kafka Connect with the MongoDB connector already installed and ready to run. You can also link:https://debezium.io/blog/2016/05/31/Debezium-on-Kubernetes/[run {prodname} on Kubernetes and OpenShift].
```
Ah, seeing this existed before, but this blog post is super-old and obsolete. Rather link to the OpenShift docs page you recently reworked (here and wherever else it's referenced).

TovaCohen replied:
I updated this in each connector file, each in a separate commit. Just pushed those commits to this PR.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mongodb.adoc:
```diff
@@ -705,6 +703,116 @@ See the {link-prefix}:{link-mongodb-connector}#mongodb-connector-properties[comp
+=== Adding connector configuration
+
+ifdef::community[]
+To run a {prodname} MongoDB connector, create a connector configuration file and add it to your Kafka Connect cluster. 
```
It's not really a file, rather a call of the REST API, as you describe further below.

TovaCohen replied:
For each connector, I updated this to:
... create a connector configuration and add the configuration to your .....
If that's not right, let me know. 
I committed and pushed these updates.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mongodb.adoc:
```diff
@@ -705,6 +703,116 @@ See the {link-prefix}:{link-mongodb-connector}#mongodb-connector-properties[comp
+
+.Prerequisites
+
+* You have Podman installed and sufficient rights to create and manage containers.
```
Podman or Docker perhaps?

TovaCohen replied:
Sure. I did this update for each connector.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mongodb.adoc:
```diff
@@ -705,6 +703,116 @@ See the {link-prefix}:{link-mongodb-connector}#mongodb-connector-properties[comp
++
+[subs=+macros]
+----
+FROM registry.redhat.io/amq7/amq-streams-kafka-25:1.5.0
```
Don't we have variable/attribute for the Strimzi (and its Kafka) version?

TovaCohen replied:
I don't see a variable in antora.yml.  We do have one downstream:
`DockerKafkaConnect: registry.redhat.io/amq7/amq-streams-kafka-25:1.5.0`
I am adding that to antora.yml with the next commit. I hope the different attribute name case is okay.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mongodb.adoc:
```diff
@@ -705,6 +703,116 @@ See the {link-prefix}:{link-mongodb-connector}#mongodb-connector-properties[comp
++
+Before Kafka Connect starts running the connector, Kafka Connect loads any third-party plug-ins that are in the `/opt/kafka/plugins` directory.
+
+.. Build the docker container image. For example, if you saved the docker file that you created in the previous step as `debezium-container-for-mongodb`, then you would run the following command:
```
docker container image -> container image.
docker file -> Dockerfile (that's the actual file name)?

TovaCohen replied:
I made these updates in the doc for each connector. Committed and pushed.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -1741,7 +1741,7 @@ endif::community[]
 With link:https://zookeeper.apache.org[Zookeeper], link:http://kafka.apache.org/[Kafka], and {link-kafka-docs}.html#connect[Kafka Connect] installed, the remaining tasks to deploy a {prodname} PostgreSQL connector are to download the link:https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/{debezium-version}/debezium-connector-postgres-{debezium-version}-plugin.tar.gz[connector ...
 
-If you need immutable containers, see link:https://hub.docker.com/r/debezium/[{prodname}'s Container images] for Zookeeper, Kafka, PostgreSQL and Kafka Connect with the PostgreSQL connector already installed and ready to run. You can also link:/blog/2016/05/31/Debezium-on-Kubernetes/[run {prodname} on Kubernetes and OpenShift].
+If you need immutable containers, see link:https://hub.docker.com/r/debezium/[{prodname}'s Container images] for Zookeeper, Kafka, PostgreSQL and Kafka Connect with the PostgreSQL connector already installed and ready to run. You can also link:https://debezium.io/blog/2016/05/31/Debezium-on-Kubernetes/[run {prodname} on Kubernetes and OpenShift].
```
See above.

TovaCohen replied:
Updated for each connector. Committed and pushed.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/sqlserver.adoc:
```diff
@@ -1208,51 +1208,46 @@ The `connect.decimal.precision` schema parameter contains an integer representin
+With link:https://zookeeper.apache.org[Zookeeper], http://kafka.apache.org/[Kafka], and {link-kafka-docs}.html#connect[Kafka Connect] installed, the remaining tasks to deploy a {prodname} SQL Server connector are to download the https://repo1.maven.org/maven2/io/debezium/debezium-connector-sqlserver/{debezium-version}/debezium-connector-sqlserver-{debezium-version}-plugin.tar.gz[connector's plug- ...
+Restart your Kafka Connect process to pick up the new JAR files.
+
+If you need immutable containers, see link:https://hub.docker.com/r/debezium/[{prodname}'s container images] for Zookeeper, Kafka, SQL Server and Kafka Connect with the SQL Server connector already installed and ready to run. You can also link:https://debezium.io/blog/2016/05/31/Debezium-on-Kubernetes/[run {prodname} on Kubernetes and OpenShift].
```
So I find the "If you need" sounds a bit odd. Perhaps "If you're working with..."? What's really needed in life, after all :)

TovaCohen replied:
Sure!  Updated in the recent commits, which reflect your comments, and which I just pushed to this PR.

gunnarmorling reviewed (commented):
Nice work! A few comments inline; most of them should apply to each occurrence of that part in the different connectors' docs.

TovaCohen commented:
Note that for the Db2, MySQL, and SQL Server connector doc, I also changed the format that explains the connector configuration settings. For those connectors only, the rendered downstream user guide had duplicated callout number in the explanatory text below the code samples. I tried many things to fix this, with no luck. So I removed the problem text and replaced it with a table.

Naros commented on documentation/antora.yml:
```diff
@@ -10,6 +10,7 @@ asciidoc:
     debezium-dev-version: '1.3'
     debezium-kafka-version: '2.6.0'
     debezium-docker-label: '1.2'
+    DockerKafkaConnect: registry.redhat.io/amq7/amq-streams-kafka-25:1.5.0
```
Isn't this variable only used in ifdef blocks related to downstream?

TovaCohen replied:
Yes, I think that is correct. 
Is there a problem with having it here?
If yes, we don't need it in the antora file.

Naros replied:
There isn't a problem having it other than the _maintenance_ aspect.

TovaCohen replied:
Please do whatever you think best.
If you want me to do something, please let me know. 
As soon as this PR is merged and backported to 1.2, I will fetch the updates and rebuild the downstream UG.

Naros commented:
LGTM, merged to master and branch 1.2
